jenkins/packer/packer-build.jenkinsfile

```diff
@@ -0,0 +1,80 @@
+            trim: true
+        )
+    }
+    stages {
+        stage('Parameters Check') {
+            steps {
+                script {
+                    if (PACKER_AGENT_JSON_NAME.isEmpty()) {
+                        currentBuild.result = 'ABORTED'
+                        error("PACKER_AGENT_JSON_NAME parameter cannot be empty.")
+                    }
+                    sh ("curl -SLO \"https://raw.githubusercontent.com/opensearch-project/opensearch-ci/main/packer/$PACKER_AGENT_JSON_NAME\"")
+                    file_validation = sh (
+                        script: "cat $PACKER_AGENT_JSON_NAME | grep \"404: Not Found\"",
+                        returnStatus: true
+                    )
+                    if (file_validation == 0) {
+                        currentBuild.result = 'ABORTED'
+                        error("PACKER_AGENT_JSON_NAME $PACKER_AGENT_JSON_NAME is not valid.")
+                    }
+                    print("Passing parameter checks.")
+                }
+            }
+        }
+        stage('packer-build') {
+            environment {
+                PACKER_BUILD_GIT_REPOSITORY_REFERENCE = "main"
+                PACKER_BUILD_GIT_REPOSITORY = "https://github.com/opensearch-project/opensearch-ci"
+            }
+            steps {
+                script {
+                    print("Getting $PACKER_AGENT_JSON_NAME from $PACKER_BUILD_GIT_REPOSITORY repository with reference $PACKER_BUILD_GIT_REPOSITORY_REFERENCE")
+                    checkout([$class: 'GitSCM', branches: [[name: PACKER_BUILD_GIT_REPOSITORY_REFERENCE ]], userRemoteConfigs: [[url: PACKER_BUILD_GIT_REPOSITORY ]]])
+                    def packer_jenkins_agent_json = readJSON file: "./packer/$PACKER_AGENT_JSON_NAME"
+                    withCredentials([
+                        string(credentialsId: "jenkins-agent-vpc-id", variable: 'vpc_id'),
+                        string(credentialsId: "jenkins-agent-subnet-id", variable: 'subnet_id'),
+                        string(credentialsId: "jenkins-agent-sg-id", variable: 'sg_id')
+                    ]){
+                        packer_jenkins_agent_json["variables"]["build-vpc"] = vpc_id
+                        packer_jenkins_agent_json["variables"]["build-subnet"] = subnet_id
+                        packer_jenkins_agent_json["variables"]["build-secgrp"] = sg_id
+                    }
+                    writeJSON file: "./packer/$PACKER_AGENT_JSON_NAME", json: packer_jenkins_agent_json, pretty: 2
+
+                    //Start to build packer
+                    withCredentials([string(credentialsId: 'jenkins-aws-account-public', variable: 'AWS_ACCOUNT_PUBLIC')]) {
+                        withAWS(role: 'opensearch-bundle', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
```

Comment:
What do we need this role for?

Reply:
To start a new agent within the same SG group to trigger the packer build.

Reply:
Need a new role with these permissions:
https://developer.hashicorp.com/packer/plugins/builders/amazon#iam-task-or-instance-role